src/api/Controllers/CodeListController.cs

```diff
@@ -40,22 +40,33 @@ public async Task<IEnumerable<Codelist>> GetAsync(string? schema = null, int? te
             codeLists = codeLists.Where(c => c.Schema == schema);
         }
 
-        if (testKindId.HasValue)
+        if (!string.IsNullOrEmpty(testKindIds))
         {
-            // Get the Geolcode associated with the TestKindId.
-            int testKindGeolCode = context.Codelists.SingleOrDefault(c => c.Id == testKindId)?.Geolcode ?? 0;
-
-            // Get the lists of Geolcodes from the HydroCodeLookup based on the testKindGeolCode.
-            List<int> hydrotestResultGeolcodes = HydroCodeLookup.HydrotestResultParameterOptions.TryGetValue(testKindGeolCode, out List<int>? tempHRIds) ? tempHRIds : new List<int>();
-            List<int> flowDirectionGeolCodes = HydroCodeLookup.HydrotestFlowDirectionOptions.TryGetValue(testKindGeolCode, out List<int>? tempFDIds) ? tempFDIds : new List<int>();
-            List<int> evaluationMethodIds = HydroCodeLookup.HydrotestEvaluationMethodOptions.TryGetValue(testKindGeolCode, out List<int>? tempEMIds) ? tempEMIds : new List<int>();
-
-            // Return the Codelists where the Codelist's Geolcode matches any of the compatible geolcodes form  the HydroCodeLookup.
-            codeLists = codeLists.Where(c =>
-                c.Geolcode != null &&
-                ((c.Schema == HydrogeologySchemas.HydrotestResultParameterSchema && hydrotestResultGeolcodes.Contains(c.Geolcode.Value)) ||
-                (c.Schema == HydrogeologySchemas.FlowdirectionSchema && flowDirectionGeolCodes.Contains(c.Geolcode.Value)) ||
-                (c.Schema == HydrogeologySchemas.EvaluationMethodSchema && evaluationMethodIds.Contains(c.Geolcode.Value))));
+            var kindIds = testKindIds.Split(',').Select(int.Parse).ToList();
```

Comment:
Q: Können die Ids auch als `int[]` oder `List<int>` übergeben werden, damit nicht von Hand geparst werden muss?

Reply:
Im Client muss `fetchApiV2` entsprechend angepasst werden damit ASP.Net das array versteht. Folgende Stackoverflow Seite hilft vielleicht: https://stackoverflow.com/questions/3450878/how-do-i-pass-a-list-of-integers-to-an-mvc-action